Sources/TuistGenerator/Mappers/GenerateModuleMapProjectMapper.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+import XcodeProj
+
+/// A project mapper that generates derived modulemap for targets which:
+/// - don't define a modulemap
+/// - contain a single header with a name different from the target name.
```

Comment:
We leverage module maps for some our mixed language modules, they usually contain several more headers, are these conditions specific to Swift Packages dependencies?

If so, perhaps the component could be renamed to `SwiftPackagesModuleMap...`

Reply:
It is a sort of "automatically generated" module map, for the simple case in which your project has a single header file.
in case you have multiple header files, or you need a custom module map, you can still define your own module map and this mapper will be a noop.

Makes sense? 🙏

Reply:
sure, was just curious as to the use-case when this may occur.

Possibly `SingleHeaderModuleMap...` - but can be renamed later if another mapper is introduced for module maps

Reply:
An example is for the SPM package linked in the description: https://github.com/mattgallagher/CwlCatchException

It contains a [CwlCatchException.h](https://github.com/mattgallagher/CwlCatchException/blob/master/Sources/CwlCatchExceptionSupport/include/CwlCatchException.h) header inside the `CwlCatchExceptionSupport` target.

Given the name of the headers differs from the target name, Xcode doesn't automatically merge it without a modulemap.
Building it with SPM instead can handle it correctly, so we have added the logic to support this automatic map also here.

(cc @pepibumur in case you have anything to add as we discussed it together)